### DISCUSSION TITLE
Point process for time-series data

### DISCUSSION THREAD

**Discussion by JacopoCerri7** [inlabru-org/inlabru#244] [General]
Hi you all,

I am currently modeling the temporal evolution of events in wildlife ecology. These events might be camera trapping records, whose temporal distribution is usually deemed to reflect activity rhythms. Conventionally, these data are modeled through kernels (e.g., https://cran.r-project.org/web/packages/overlap/index.html). However the use of kernels is purely descriptive, and they do not allow to include covariates to explain the density of records in time (e.g. temperature).
Do you know if it is possible to fit time-series point process in INLA, perhaps with inlabru, and to account for covariates in these models, through a linear term or a spline ? In case do you have any example for this? I had a look around but I only found examples for spatial or spatiotemporal point processes.

Best,

Jacopo Cerri

**Comment by finnlindgren** [comment]
Time series point processes are implemented in the same way as spatial ones; but I suspect you mean general temporal process, rather than a discretely indexed model? Both can be done; it's all about what you specify as the `domain` when calling `like()` or `fm_int()`
``` r
# Discrete time:
like(..., domain = list(time = 1:T))
# Continuous time:
like(..., domain = list(time = temporal_mesh))
```
The continuous time using a temporal mesh (`fm_mesh_1d()`) works the same as with `fm_mesh_2d()`, except that the corresponding model component needs to get time as input instead of spatial geometry.

**Reply by JacopoCerri7** [reply]
Thanks !

I had a look at the fm_mesh_1d command. Do you have any advice on how to specify the mesh? I found many examples for bidimensional meshes in spatial point process, but only this 1-d meshes: 

https://cran.r-project.org/web/packages/fmesher/vignettes/basic_use.html

I saw that there are many specification for boundaries(e.g., 'neumann', 'dirichlet', 'free', 'cyclic') and I do not know exactly to what they correspond and on which basis I can select them. Sorry for these questions, but I am a rather applied scientist and my background in statistics is a bit lacking on this topic.

Best,

Jacopo Cerri

**Reply by finnlindgren** [reply]
I'd need more information to answer that; every problem is different, and can require different approaches. If you're familiar with the 2d version, then the 1d is "the same but simpler". You give the spline knots for the d representation (same as 2d vertices if degree=1 (default), but it also handles piecewise quadratics with degree=2). Then you use it the same way, likely with boundary="neumann", but just as in 2d, you then may want to extend the domain a bit in the start and end.
``` r
mydata <- runif(1000) * 100 + 10 # Data between 10 and 110
# Want to model on (10,110), extending by 20 units each direction, for example:
mesh <- fm_mesh_1d(seq(-10, 130, by = 5), degree = 2, boundary = "neumann")
```

- neumann = derivative is zero at start and end
- dirichlet = value is zero at start and end
- cyclic = the process wraps around to the beginning (useful when the coordinate is e.g. an angle, or "month of the year")
- free = no boundary constraints at all (give a rank-deficient instrinsic stationary model)
